### PR TITLE
Add a RELIC package and unit test that shows how to use it

### PR DESCRIPTION
**Pull request by OlegHahm**
Add install of cmake to .travis.yml.

Test-Information:

This has been tested on OS X and on Ubuntu for both the native target
and STM32F4Discovery.

Adopted #2213 from @tfar.

### CODE REVIEW

**Comment by haukepetersen** [comment]
I can't compile under Mint 17:
```
CMake Error at CMakeLists.txt:2 (cmake_minimum_required):
  CMake 3.1 or higher is required.  You are running version 2.8.12.2
```
So Ubuntu users (and Travis?) are probably running into the same problem...

**Comment by OlegHahm** [comment]
Can you try to manually revert this commit: https://github.com/relic-toolkit/relic/commit/9969f1beb22d4242f89e761398a1a5c6cb4cbed5 and see if it works?

**Comment by OlegHahm** [comment]
@haukepetersen, now it should work.

**Comment by OlegHahm** [comment]
Opened an issue concerning the AVR and MSP430 platforms at Relic's upstream repository: https://github.com/relic-toolkit/relic/issues/27

File: tests/unittests/tests-relic/tests-relic.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        bn_print(privateB);
+        printf("\npublic key:  ");
+        ec_print(publicB);
+        printf("\n");
+#endif
+
+        /* In a protocol you would exchange the public keys now */
+
+        /* User A calculates shared secret */
+        assert(cp_ecdh_key(sharedKeyA, MD_LEN, privateA, publicB) == STS_OK);
```
Comment (tfar) [inline tests/unittests/tests-relic/tests-relic.c]: @OlegHahm Please replace this line with
```
TEST_ASSERT_EQUAL_INT(STS_OK, cp_ecdh_key(sharedKeyA, MD_LEN, privateA, publicB));
```

The assert RIOT defines results in complete removal of that line in the unit testing configuration. This resulted in the tests failing.

File: tests/unittests/tests-relic/tests-relic.c [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+#if (TEST_RELIC_SHOW_OUTPUT == 1)
+        printf("\nshared key computed by user A: ");
+        print_mem(sharedKeyA, MD_LEN);
+#endif
+
+        /* User B calculates shared secret */
+        assert(cp_ecdh_key(sharedKeyB, MD_LEN, privateB, publicA) == STS_OK);
```
Comment (tfar) [inline tests/unittests/tests-relic/tests-relic.c]: @OlegHahm Please replace this line with
```
TEST_ASSERT_EQUAL_INT(STS_OK, cp_ecdh_key(sharedKeyB, MD_LEN, privateB, publicA));
```

File: tests/unittests/tests-relic/tests-relic.c (hunk 1 above)
Comment (OlegHahm) [inline tests/unittests/tests-relic/tests-relic.c]: The RIOT definition of assert if building with `NDEBUG` is exactly the same as on Linux and apparently unittests are built with NDEBUG.

**Comment by OlegHahm** [comment]
@tfar, thanks for spotting these unittests errors. Everybody okay to squash?

**Comment by OlegHahm** [comment]
Travis complains about mixing spaces with tabs:
```
pkg/relic/0001-fixed-signedness-of-counter-variable.patch:17: space before tab in indent.
+ 	bn_t m, t, z;
pkg/relic/0001-fixed-signedness-of-counter-variable.patch:22: space before tab in indent.
+ 	size = bn_size_bin(prv->n);
pkg/relic/0001-fixed-signedness-of-counter-variable.patch:26: new blank line at EOF.
pkg/relic/0002-don-t-redefine-ALIGN.patch:69: new blank line at EOF.
pkg/relic/0003-require-only-CMake-version-2.8.patch:23: new blank line at EOF.
ERROR: This change introduces new whitespace errors
```
Honestly, no idea how to fix this. I think replacing tabs by spaces is out of question.

**Comment by cgundogan** [comment]
I'd say just remove both whitespaces in front of the tabs. And remove the last line of the patch file. This should fix it.

**Comment by OlegHahm** [comment]
I think the whitespaces are required for the patch format.

**Comment by OlegHahm** [comment]
Yepp: `fatal: corrupt patch at line 12`

**Comment by cgundogan** [comment]
I mean you could remove the white space or repace the tab with white spaces and then create the patch again. Deleting the last line from the patch file should be safe, though.

**Comment by OlegHahm** [comment]
> I mean you could remove the white space

There is no white space in the code. The white space is added by the patch format.

>  or replace the tab with white spaces 

As I stated above: I don't think this is a good idea: Either we would end up with one file that uses spaces instead of tabs in one line which would be very odd or we need to change all tabs to white spaces which seems to be pretty stupid.

**Comment by OlegHahm** [comment]
Getting https://github.com/relic-toolkit/relic/pull/26 merged would be my preferred solution.

**Comment by kaspar030** [comment]
Same patch problem as my microcoap patch. IMHO we should ignore whitespace error messages generated for patch files.

**Comment by OlegHahm** [comment]
It's not quite the same problem IIRC, but I agree that ignoring this is probably the best way forward.

**Comment by kaspar030** [comment]
Marking *.patch as binary in .gitattributes makes git not complain about the whitespace problems.

**Comment by OlegHahm** [comment]
Good idea: #4402.

**Comment by kaspar030** [comment]
#4403

**Comment by kaspar030** [comment]
damn :)

**Comment by OlegHahm** [comment]
Should I squash this one?

**Comment by tfar** [comment]
@OlegHahm Went through all commits and reviewed them. Looks good to me.

**Comment by OlegHahm** [comment]
Thanks for the review. I squashed. I guess this was an ACK then?

**Comment by haukepetersen** [comment]
sorry for the delay: can confirm, works good under Mint17 now. Looks good to me, ACK from my side.

**Comment by OlegHahm** [comment]
Thx!

**Pull request merged**